dev/src/index.ts

```diff
@@ -504,14 +504,24 @@ export class Firestore implements firestore.Firestore {
 
     if (settings.host !== undefined) {
       validateHost('settings.host', settings.host);
-      if (settings.servicePath !== undefined) {
-        throw new Error(
-          'Cannot set both "settings.host" and "settings.servicePath".'
+      if (
+        settings.servicePath !== undefined &&
+        settings.servicePath !== settings.host
+      ) {
+        // eslint-disable-next-line no-console
```

Comment:
Not sure about the error message here, since the host is stored on `settings.servicePath`. Could we refer to hosts set on `settings.servicePath` and `settings.apiEndpoint` as "existing hosts"? That way, we could simply and combine the two error messages into something like "settings.host does not match the existing host..."